libpermian/plugins/anaconda_webui/hypervisor.py

```diff
@@ -0,0 +1,140 @@
+import subprocess
+import time
+import logging
+import os
+import tempfile
+import random
+
+
+START_PORT = '42000'
+LOGGER = logging.getLogger(__name__)
+
+
+class Hypervisor():
+    def __init__(self, host):
+        self.host = host
+
+        if host in ['localhost', '127.0.0.1', '::1']:
+            self.remote = False
+            self.qemu_host = 'qemu:///system'
+        else:
+            self.remote = True
+            self.qemu_host = f'qemu+ssh://{host}/system'
+            self.tempdir = tempfile.TemporaryDirectory(prefix='permian_sshctl.')
+            self.ssh_ctl_file = os.path.join(self.tempdir.name, '%r@%h:%p')
+            self._open_connection()
+
+    def setup(self):
+        # Try to create storage pool boot-scratch to avoid issues later, if it exist its ok for this command to fail.
+        self.virsh_call([
+            'pool-create-as', '--name', 'boot-scratch',
+            '--type', 'dir', '--target' ,'/var/lib/libvirt/boot',
+            '--build'
+        ])
+
+    def forwarding_cleanup(self):
```

Comment:
Similarly like in case of `_close_connection` it would be better if this method was also called implicitly a.g. in `__del__`.

Or even better if there was some kind of `forwarding` method that could be used in `with` statement that would call this method on exit from the `with` statement, but such a change would require bigger refactoring that could be done later and does not need to be introduced in this phase.

Reply:
I don't know if it is good idea to rely on __del__ for the cleanup

Reply:
This method calls `self._close_connection()`, which calls `subprocess.run()` - I don't think python guarantees that the `subprocess` modul is still usable when `__del__` is called. We saw such problem with the `logging` module elsewhere, so I would avoid calling this method in `__del__`.

Reply:
Looks good to me now.